pkg/activator/handler/healthz_handler_test.go

```diff
@@ -77,3 +77,26 @@ func TestHealthHandler(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkHealthHandler(b *testing.B) {
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	handler := HealthHandler{HealthCheck: func() error { return nil }, NextHandler: baseHandler, Logger: ktesting.TestLogger(b)}
+	req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
+	req.Header.Set("User-Agent", "kube-probe/something")
```

Comment:
Same here: Can we do one test with and one without the agent set, to be able to compare overhead in either situation and to assure that the general case (not a probe) is super small.